### PR TITLE
DOC: add release notes link to PyPI.

### PR DESCRIPTION
**Pull request by DanShatford**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -66,6 +66,7 @@ documentation = "https://numpy.org/doc/"
 source = "https://github.com/numpy/numpy"
 download = "https://pypi.org/project/numpy/#files"
 tracker = "https://github.com/numpy/numpy/issues"
+"Release Notes" = "https://numpy.org/doc/stable/release"
```
Comment (rgommers) [inline pyproject.toml:69]: This does seem to be supported by PyPI and yield an icon: https://github.com/pypi/warehouse/blob/d12deea005f6652862ad0b9d28ca8c3d39bc7402/warehouse/templates/packaging/detail.html#L24

Comment (rgommers) [reply pyproject.toml:69]: Can you make this all lower case though? This fits better with the other content in this file.

File: setup.py [hunk 2]
```diff
@@ -513,6 +513,7 @@ def setup_package():
         project_urls={
             "Bug Tracker": "https://github.com/numpy/numpy/issues",
             "Documentation": get_docs_url(),
+            "Release Notes" = "https://numpy.org/doc/stable/release",
```
Comment (rgommers) [inline setup.py:516]: Can you please drop this change? This won't be used and will give a merge conflict in another PR.

**Review by rgommers — COMMENTED** [review]
Thanks @DanShatford. This seems like a useful link to add. A few minor comments.

**Comment by DanShatford** [comment]
thanks for looking this over. made those modifications now.

**Comment by charris** [comment]
Thanks @DanShatford .

**Pull request merged**